client/src/components/Meetup/Meetup.js

```diff
@@ -1,8 +1,689 @@
+  const [attendees, setAttendees] = useState([]);
+  const [pendingAttendees, setPendingAttendees] = useState([]);
+
+  const locale = "en-US";
+  const meetupTimeOptions = {
+    hour: "numeric",
+    minute: "numeric",
+    day: "numeric",
+    month: "long",
+    year: "numeric",
+    timeZoneName: "short",
+  };
+  const userDateOptions = {
+    day: "numeric",
+    month: "long",
+    year: "numeric",
+  };
+
+  async function setAttendeeLists(attendeeList, pendingAttendeesList) {
+    setAttendees(await fetcher.getDataForUsers(attendeeList));
+    setPendingAttendees(await fetcher.getDataForUsers(pendingAttendeesList));
+  }
+
+  const setUserStatus = (userData, meetupData) => {
+    if (userData && meetupData.owner === userData.id) {
+      setUserMeetupStatus("owner");
+    } else if (
+      meetupData.pendingAttendees &&
+      meetupData.pendingAttendees.includes(userData.id)
+    ) {
+      setUserMeetupStatus("pending");
+    } else if (
+      meetupData.attendees &&
+      meetupData.attendees.includes(userData.id)
+    ) {
+      setUserMeetupStatus("attending");
+    } else if (meetupData.rejected) {
+      setUserMeetupStatus("rejected");
+    }
+  };
+
+  useEffect(async () => {
+    const { res: userRes, resJSON: userJSON } = await fetcher.getUserData();
+    if (!userRes.ok) {
+      setIsLoading(false);
+      return;
+    }
+    const { res: meetupRes, resJSON: meetupJSON } = await fetcher.getMeetup(id);
+    if (!meetupRes.ok) {
+      setFetchMeetupError(meetupRes.status);
+      setIsLoading(false);
+      return;
+    }
+    const { res: ownerRes, resJSON: ownerJSON } = await fetcher.getUserData(
+      meetupJSON.owner
+    );
+    if (!ownerRes.ok) {
+      setIsLoading(false);
+      return;
+    }
+
+    setUser(userJSON);
+    setEventDetails({
+      description: meetupJSON.description,
+      location: meetupJSON.location,
+      maxCapacity: meetupJSON.maxCapacity,
+      minCapacity: meetupJSON.minCapacity,
+      owner: ownerJSON,
+      tags: meetupJSON.tags,
+      time: new Date(meetupJSON.time),
+      title: meetupJSON.title,
+    });
+    setAttendeeLists(meetupJSON.attendees, meetupJSON.pendingAttendees);
+    setUserStatus(userJSON, meetupJSON);
+    setIsLoading(false);
+  }, []);
+
+  const Spinner = (
+    <Grid
+      container
+      spacing={3}
+      justify="center"
+      alignItems="center"
+      className={classes.spinner}
+    >
+      <CircularProgress />
+    </Grid>
+  );
+
+  const renderError = (status) => {
+    let errorMessage = "";
+
+    switch (status) {
+      case 404:
+        errorMessage = "Error: Meetup not found.";
+        break;
+      case 400:
+        errorMessage = "Error: Bad request.";
+        break;
+      default:
+        errorMessage = "Error: Unspecified error occured.";
+    }
+
+    return (
+      <Grid container spacing={1}>
+        <Typography className={classes.error}>{errorMessage}</Typography>
```

Comment:
We don't need a class here.

```suggestion
        <Typography color="error">{errorMessage}</Typography>
```